### ISSUE TITLE
Install for openshift-logging fails after rolling out elasticsearch pods twice

### ISSUE THREAD

**Issue by odbawler** [openshift/openshift-ansible#11018]
#### Description

Only recently has this started happening, this used to install as expected. When installing openshift-logging, the deployment fails when rolling out elasticsearch pods. 

For some reason, it's performing the rollout task `Rolling out new pod(s) for {{ _es_node }}]` twice. The first time it is successful, and I can see the ES pod running in my OpenShift cluster. However when it tries the second time (which I don't think it should be doing?), I get an error:
`error: #1 is already in progress (Running).`

I have observed more strange behaviour as it also deploys kibana and `logging-kibana-ops`, even when I specify `openshift_logging_use_ops=false`. 

##### Observed Results
A subset of the -vvv output from exeuting the openshift-logging/config.yml playbook (just included some of the task names to display the task flow):

```
RUNNING HANDLER [openshift_logging_elasticsearch : Rolling out new pod(s) for {{ _es_node }}] *******************************************************************************************************************
task path: /Users/User1/git/ansible-platform/.vendor/openshift-ansible/roles/openshift_logging_elasticsearch/tasks/restart_es_node.yml:17
changed: [master.local] => {
    "changed": true, 
    "cmd": [
        "/usr/local/bin/oc", 
        "--config=/etc/origin/master/admin.kubeconfig", 
        "rollout", 
        "latest", 
        "logging-es-data-master-deagjg9n", 
        "-n", 
        "openshift-logging"
    ], 
    "delta": "0:00:00.228193", 
    "end": "2019-01-16 08:39:41.505036", 
    "invocation": {
        "module_args": {
            "_raw_params": "/usr/local/bin/oc --config=/etc/origin/master/admin.kubeconfig rollout latest logging-es-data-master-deagjg9n -n openshift-logging", 
            "_uses_shell": false, 
            "argv": null, 
            "chdir": null, 
            "creates": null, 
            "executable": null, 
            "removes": null, 
            "stdin": null, 
            "warn": true
        }
    }, 
    "rc": 0, 
    "start": "2019-01-16 08:39:41.276843", 
    "stderr": "", 
    "stderr_lines": [], 
    "stdout": "deploymentconfig.apps.openshift.io/logging-es-data-master-deagjg9n rolled out", 
    "stdout_lines": [
        "deploymentconfig.apps.openshift.io/logging-es-data-master-deagjg9n rolled out"
    ]
}

RUNNING HANDLER [openshift_logging_elasticsearch : Waiting for {{ _es_node }} to finish scaling up]

RUNNING HANDLER [openshift_logging_elasticsearch : set_stats] 

RUNNING HANDLER [openshift_logging_elasticsearch : Enable shard balancing for logging-{{ _cluster_component }} cluster]

RUNNING HANDLER [openshift_logging_elasticsearch : Evaluating status of rolled out pod] *************************************************************************************************************************

RUNNING HANDLER [openshift_logging_elasticsearch : Waiting for ES node {{ _es_node }} health to be in ['green']]

RUNNING HANDLER [openshift_logging_elasticsearch : set_stats]

RUNNING HANDLER [openshift_logging_elasticsearch : Evaluating cluster health]

RUNNING HANDLER [openshift_logging_elasticsearch : Enable shard balancing for logging-{{ _cluster_component }} cluster]

RUNNING HANDLER [openshift_logging_elasticsearch : Cleaning up temp dir]


RUNNING HANDLER [openshift_logging_elasticsearch : command]

RUNNING HANDLER [openshift_logging_elasticsearch : command]

###################################
###################################
<<<< it appears to loop here? >>>>
###################################
###################################

RUNNING HANDLER [openshift_logging_elasticsearch : Exporting secrets to use communicating with the ES cluster]

RUNNING HANDLER [openshift_logging_elasticsearch : Checking current health for {{ _es_node }} cluster]

RUNNING HANDLER [openshift_logging_elasticsearch : Set Logging message to manually restart]

RUNNING HANDLER [openshift_logging_elasticsearch : debug]

RUNNING HANDLER [openshift_logging_elasticsearch : command]

RUNNING HANDLER [openshift_logging_elasticsearch : Performing full cluster restart for {{ _cluster_component }} cluster]

RUNNING HANDLER [openshift_logging_elasticsearch : Performing rolling cluster restart for {{ _cluster_component }} cluster] 

RUNNING HANDLER [openshift_logging_elasticsearch : Flushing for logging-{{ _cluster_component }} cluster]

RUNNING HANDLER [openshift_logging_elasticsearch : Disable shard balancing for logging-{{ _cluster_component }} cluster]

RUNNING HANDLER [openshift_logging_elasticsearch : include_tasks]

RUNNING HANDLER [openshift_logging_elasticsearch : Disable shard balancing for logging-{{ _cluster_component }} cluster] 

RUNNING HANDLER [openshift_logging_elasticsearch : Rolling out new pod(s) for {{ _es_node }}] *******************************************************************************************************************
task path: /Users/User1/git/ansible-platform/.vendor/openshift-ansible/roles/openshift_logging_elasticsearch/tasks/restart_es_node.yml:17
fatal: [master.local]: FAILED! => {
    "changed": true, 
    "cmd": [
        "/usr/local/bin/oc", 
        "--config=/etc/origin/master/admin.kubeconfig", 
        "rollout", 
        "latest", 
        "logging-es-data-master-deagjg9n", 
        "-n", 
        "openshift-logging"
    ], 
    "delta": "0:00:00.225054", 
    "end": "2019-01-16 08:40:02.763924", 
    "invocation": {
        "module_args": {
            "_raw_params": "/usr/local/bin/oc --config=/etc/origin/master/admin.kubeconfig rollout latest logging-es-data-master-deagjg9n -n openshift-logging", 
            "_uses_shell": false, 
            "argv": null, 
            "chdir": null, 
            "creates": null, 
            "executable": null, 
            "removes": null, 
            "stdin": null, 
            "warn": true
        }
    }, 
    "msg": "non-zero return code", 
    "rc": 1, 
    "start": "2019-01-16 08:40:02.538870", 
    "stderr": "error: #1 is already in progress (Running).", 
    "stderr_lines": [
        "error: #1 is already in progress (Running)."
    ], 
    "stdout": "", 
    "stdout_lines": []
}
```

##### Additional Information

```
[masters]
master.local openshift_kubelet_name_override=master.local openshift_node_group_name=node-config-master

[etcd]
master.local openshift_kubelet_name_override=master.local openshift_node_group_name=node-config-master

[nodes]
master.locall openshift_kubelet_name_override=master.local openshift_node_group_name=node-config-master
infra.local openshift_kubelet_name_override=infra.local openshift_node_group_name=node-config-infra
node1.local openshift_kubelet_name_override=node1.local openshift_node_group_name=node-config-build
node2.local openshift_kubelet_name_override=node2.local openshift_node_group_name=node-config-compute
node3.local openshift_kubelet_name_override=node3.local openshift_node_group_name=node-config-compute

[OSEv3:children]
masters
etcd
nodes

[OSEv3:vars]
ansible_user = centos
ansible_become = yes
openshift_deployment_type = origin
openshift_release = v3.11
debug_level = 2
openshift_disable_check = memory_availability,disk_availability
openshift_node_groups = [{'name': 'node-config-master', 'labels': ['node-role.kubernetes.io/master=true', 'region=master']}, {'name': 'node-config-infra', 'labels': ['node-role.kubernetes.io/infra=true', 'region=infra']}, {'name': 'node-config-compute', 'labels': ['node-role.kubernetes.io/compute=true', 'region=primary']}, {'name': 'node-config-build', 'labels': ['node-role.kubernetes.io/compute=true', 'region=build', 'role=build']}]
openshift_cloudprovider_kind = aws
openshift_storageclass_parameters = {'type': 'gp2', 'encrypted': 'true'}
openshift_master_api_port = 8443
openshift_master_console_port = 8443
openshift_image_tag = v3.11.0
openshift_hosted_router_selector = 'region=infra'
os_sdn_network_plugin_name = 'redhat/openshift-ovs-multitenant'
openshift_hosted_registry_selector = 'region=infra'
openshift_hosted_manage_registry = true
openshift_hosted_manage_registry_console = true
openshift_hosted_registry_storage_kind = object
openshift_hosted_registry_storage_provider = s3
openshift_hosted_registry_storage_s3_encrypt = true
openshift_metrics_install_metrics = true
openshift_metrics_image_tag = "v3.9"
openshift_metrics_cassandra_image = "docker.io/openshift/origin-metrics-cassandra:{{ openshift_metrics_image_tag }}"
openshift_metrics_hawkular_agent_image = "docker.io/openshift/origin-metrics-hawkular-openshift-agent:{{ openshift_metrics_image_tag }}"
openshift_metrics_hawkular_metrics_image = "docker.io/openshift/origin-metrics-hawkular-metrics:{{ openshift_metrics_image_tag }}"
openshift_metrics_schema_installer_image = "docker.io/openshift/origin-metrics-schema-installer:{{ openshift_metrics_image_tag }}"
openshift_metrics_heapster_image = "docker.io/openshift/origin-metrics-heapster:{{ openshift_metrics_image_tag }}"
openshift_metrics_storage_kind = dynamic
openshift_logging_install_logging = true
openshift_logging_image_tag = v3.11
openshift_logging_elasticsearch_image = "docker.io/openshift/origin-logging-elasticsearch:{{ openshift_logging_image_tag }}"
openshift_logging_curator_image = "docker.io/openshift/origin-logging-curator:{{ openshift_logging_image_tag }}"
openshift_logging_eventrouter_image = "docker.io/openshift/origin-logging-eventrouter:{{ openshift_logging_image_tag }}"
openshift_logging_fluentd_image = "docker.io/openshift/origin-logging-fluentd:{{ openshift_logging_image_tag }}"
openshift_logging_kibana_image = "docker.io/openshift/origin-logging-kibana:{{ openshift_logging_image_tag }}"
openshift_logging_mux_image = "docker.io/openshift/origin-logging-mux:{{ openshift_logging_image_tag }}"
openshift_logging_kibana_proxy_image = "docker.io/openshift/origin-logging-auth-proxy:{{ openshift_logging_image_tag }}"
openshift_logging_es_memory_limit = 2Gi
openshift_logging_es_pvc_dynamic = true
openshift_logging_es_pvc_size = 40G
openshift_logging_storage_kind = dynamic
openshift_logging_es_nodeselector = {"region":"infra"}
openshift_logging_kibana_nodeselector = {"region":"infra"}
openshift_logging_curator_nodeselector = {"region":"infra"}
openshift_logging_use_ops = false

```

**Comment by vrutkovs** [comment]
Please put the full ansible playbook log in a gist, running with `-vvv`. Its not clear what started these handlers twice.

/cc @jcantrill @ewolinetz

**Comment by odbawler** [comment]
[Full output](https://gist.githubusercontent.com/odbawler/d8234d58e24f697edbdbb28a53a610b0/raw/f22aa364fe24aa8f551563e29486c380504e64bc/output.log)

**Comment by odbawler** [comment]
Rolled back my ansible version from 2.7.5 to 2.6.10 and its working

**Comment by Bengrunt** [comment]
Hey, rolled back as well to ansible 2.6 and it now works as expected.

**Comment by Drakkai** [comment]
> I have observed more strange behaviour as it also deploys kibana and logging-kibana-ops, even when I specify openshift_logging_use_ops=false.
I've observed the same, did you manage to resolve this?

**Comment by Bengrunt** [comment]
> > I have observed more strange behaviour as it also deploys kibana and logging-kibana-ops, even when I specify openshift_logging_use_ops=false.
> 
> I've observed the same, did you manage to resolve this?

When I switched back to Ansible 2.6 this problem disappeared as well (aside from the ops PVC class choice).

**Comment by odbawler** [comment]
> > > I have observed more strange behaviour as it also deploys kibana and logging-kibana-ops, even when I specify openshift_logging_use_ops=false.
> > 
> > 
> > I've observed the same, did you manage to resolve this?
> 
> When I switched back to Ansible 2.6 this problem disappeared as well (aside from the ops PVC class choice).

Yep, check your Ansible version and roll back to a 2.6.x version

**Comment by sdodson** [comment]
/reopen
/cc @mtnbikenc 
/assign @jcantrill @ewolinetz 

We should figure out what's wrong with Ansible 2.7 as we'll likely need to support it when Ansible 2.6 goes EOL.

**Comment by InfoSec812** [comment]
I just ran into this as well on the release-3.11 branch. I have downgraded Ansible and hopefully that will get me installed, but other items I use require Ansible 2.7 already, so this is not a great solution going forward.

**Comment by jan-zmeskal** [comment]
@richm Isn't this fixed by https://github.com/openshift/openshift-ansible/pull/11220 ?

**Comment by richm** [comment]
> @richm Isn't this fixed by #11220 ?

Yes, looks like it

**Comment by martinbe1io** [comment]
Hi, 

I ran into #11132 and thought it should be be fixed in the latest version so I tried to install logging again with this playbook parts

```
openshift_logging_install_logging=true
openshift_logging_purge_logging=false
openshift_logging_image_tag = latest
openshift_logging_es_nodeselector={"node-role.kubernetes.io/infra":"true"}
openshift_logging_kibana_nodeselector={"node-role.kubernetes.io/infra":"true"}
openshift_logging_curator_nodeselector={"node-role.kubernetes.io/infra":"true"}
openshift_logging_storage_kind=dynamic
openshift_logging_es_pvc_size=10Gi
openshift_logging_elasticsearch_storage_type=pvc
openshift_logging_es_memory_limit=4Gi
openshift_logging_es_pvc_storage_class_name="glusterfs-registry-block"
openshift_logging_kibana_hostname=logging.apps.oc2.be1.io
```

it thought it is enough to use image tag latest but still struggling at this part in the ansible installer 
```
RUNNING HANDLER [openshift_logging_elasticsearch : Disable shard balancing for logging-{{ _cluster_component }} cluster] ***********************************************************************************
Friday 22 March 2019  16:51:24 +0100 (0:00:00.417)       0:03:31.112 ********** 
skipping: [oc2n01de.be1.io]

RUNNING HANDLER [openshift_logging_elasticsearch : Rolling out new pod(s) for {{ _es_node }}] **************************************************************************************************************
Friday 22 March 2019  16:51:25 +0100 (0:00:00.224)       0:03:31.336 ********** 
fatal: [oc2n01de.be1.io]: FAILED! => {"changed": true, "cmd": ["oc", "--config=/etc/origin/master/admin.kubeconfig", "rollout", "latest", "logging-es-data-master-jxy2mfdo", "-n", "openshift-logging"], "delta": "0:00:00.438604", "end": "2019-03-22 16:51:26.251282", "msg": "non-zero return code", "rc": 1, "start": "2019-03-22 16:51:25.812678", "stderr": "error: #1 is already in progress (Running).", "stderr_lines": ["error: #1 is already in progress (Running)."], "stdout": "", "stdout_lines": []}

RUNNING HANDLER [openshift_logging_elasticsearch : Waiting for {{ _es_node }} to finish scaling up] ********************************************************************************************************
Friday 22 March 2019  16:51:26 +0100 (0:00:01.209)       0:03:32.545 ********** 

RUNNING HANDLER [openshift_logging_elasticsearch : set_stats] **********************************************************************************************************************************************
Friday 22 March 2019  16:51:26 +0100 (0:00:00.001)       0:03:32.546 ********** 

RUNNING HANDLER [openshift_logging_elasticsearch : Enable shard balancing for logging-{{ _cluster_component }} cluster] ************************************************************************************
Friday 22 March 2019  16:51:26 +0100 (0:00:00.000)       0:03:32.547 ********** 

RUNNING HANDLER [openshift_logging_elasticsearch : Evaluating status of rolled out pod] ********************************************************************************************************************
Friday 22 March 2019  16:51:26 +0100 (0:00:00.000)       0:03:32.548 ********** 

RUNNING HANDLER [openshift_logging_elasticsearch : Waiting for ES node {{ _es_node }} health to be in ['green']] *******************************************************************************************
Friday 22 March 2019  16:51:26 +0100 (0:00:00.000)       0:03:32.549 ********** 

RUNNING HANDLER [openshift_logging_elasticsearch : set_stats] **********************************************************************************************************************************************
Friday 22 March 2019  16:51:26 +0100 (0:00:00.000)       0:03:32.550 ********** 

RUNNING HANDLER [openshift_logging_elasticsearch : Evaluating cluster health] ******************************************************************************************************************************
Friday 22 March 2019  16:51:26 +0100 (0:00:00.000)       0:03:32.551 ********** 

RUNNING HANDLER [openshift_logging_elasticsearch : Waiting for ES cluster logging{{ _cluster_component }} to be up] ****************************************************************************************
Friday 22 March 2019  16:51:26 +0100 (0:00:00.002)       0:03:32.553 ********** 

RUNNING HANDLER [openshift_logging_elasticsearch : Enable shard balancing for logging-{{ _cluster_component }} cluster] ************************************************************************************
Friday 22 March 2019  16:51:26 +0100 (0:00:00.000)       0:03:32.554 ********** 

RUNNING HANDLER [openshift_logging_elasticsearch : Cleaning up temp dir] ***********************************************************************************************************************************
Friday 22 March 2019  16:51:26 +0100 (0:00:00.001)       0:03:32.555 ********** 

RUNNING HANDLER [openshift_logging_elasticsearch : set_fact] ***********************************************************************************************************************************************
Friday 22 March 2019  16:51:26 +0100 (0:00:00.010)       0:03:32.566 ********** 

NO MORE HOSTS LEFT *****************************************************************************************************************************************************************************************

PLAY RECAP *************************************************************************************************************************************************************************************************
localhost                  : ok=12   changed=0    unreachable=0    failed=0   
oc2n01de.be1.io            : ok=253  changed=59   unreachable=0    failed=1   
oc2n02de.be1.io            : ok=1    changed=0    unreachable=0    failed=0   
oc2s01de.be1.io            : ok=1    changed=0    unreachable=0    failed=0   
oc2s02de.be1.io            : ok=1    changed=0    unreachable=0    failed=0   
oc2s03de.be1.io            : ok=1    changed=0    unreachable=0    failed=0   


INSTALLER STATUS *******************************************************************************************************************************************************************************************
Initialization   : Complete (0:00:25)
Logging Install  : In Progress (0:03:08)
	This phase can be restarted by running: playbooks/openshift-logging/config.yml
Friday 22 March 2019  16:51:26 +0100 (0:00:00.007)       0:03:32.574 ********** 
=============================================================================== 
```

intersting is that the pods seem to run perfectly 
```
[root@oc2n01de openshift-ansible]# oc get pods
NAME                                      READY     STATUS    RESTARTS   AGE
logging-es-data-master-jxy2mfdo-1-cjcdq   2/2       Running   0          20m
logging-fluentd-cpftr                     1/1       Running   0          21m
logging-fluentd-kv8q6                     1/1       Running   0          21m
logging-fluentd-rcbbx                     1/1       Running   0          21m
logging-fluentd-spx9x                     1/1       Running   0          21m
logging-fluentd-vbg42                     1/1       Running   0          21m
logging-kibana-ops-1-9smr7                2/2       Running   0          21m
```

es-datamaster pod reports cluster green
```

2019-03-22 15:51:21,924][INFO ][container.run            ] Begin Elasticsearch startup script
--
  | [2019-03-22 15:51:21,931][INFO ][container.run            ] Comparing the specified RAM to the maximum recommended for Elasticsearch...
  | [2019-03-22 15:51:21,932][INFO ][container.run            ] Inspecting the maximum RAM available...
  | [2019-03-22 15:51:21,935][INFO ][container.run            ] ES_JAVA_OPTS: ' -Xms2048m -Xmx2048m'
  | [2019-03-22 15:51:21,937][INFO ][container.run            ] Building required jks files and truststore
  | [2019-03-22 15:51:21,943][INFO ][container.run            ] Setting heap dump location /elasticsearch/persistent/heapdump.hprof
  | [2019-03-22 15:51:21,945][INFO ][container.run            ] Checking if Elasticsearch is ready
  | [2019-03-22 15:51:21,945][INFO ][container.run            ] ES_JAVA_OPTS: ' -Xms2048m -Xmx2048m -XX:HeapDumpPath=/elasticsearch/persistent/heapdump.hprof -Dsg.display_lic_none=false -Dio.netty.recycler.maxCapacityPerThread=0 -Dio.netty.allocator.type=unpooled'
  |  
  | ### LICENSE NOTICE Search Guard ###
  |  
  | If you use one or more of the following features in production
  | make sure you have a valid Search Guard license
  | (See https://floragunn.com/searchguard-validate-license )
  |  
  | * Kibana Multitenancy
  | * LDAP authentication/authorization
  | * Active Directory authentication/authorization
  | * REST Management API
  | * JSON Web Token (JWT) authentication/authorization
  | * Kerberos authentication/authorization
  | * Document- and Fieldlevel Security (DLS/FLS)
  | * Auditlogging
  |  
  | In case of any doubt mail to <sales@floragunn.com>
  | ###################################
  |  
  | ### LICENSE NOTICE Search Guard ###
  |  
  | If you use one or more of the following features in production
  | make sure you have a valid Search Guard license
  | (See https://floragunn.com/searchguard-validate-license )
  |  
  | * Kibana Multitenancy
  | * LDAP authentication/authorization
  | * Active Directory authentication/authorization
  | * REST Management API
  | * JSON Web Token (JWT) authentication/authorization
  | * Kerberos authentication/authorization
  | * Document- and Fieldlevel Security (DLS/FLS)
  | * Auditlogging
  |  
  | In case of any doubt mail to <sales@floragunn.com>
  | ###################################
  | Consider setting -Djdk.tls.rejectClientInitiatedRenegotiation=true to prevent DoS attacks through client side initiated TLS renegotiation.
  | Consider setting -Djdk.tls.rejectClientInitiatedRenegotiation=true to prevent DoS attacks through client side initiated TLS renegotiation.
  | SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
  | SLF4J: Defaulting to no-operation (NOP) logger implementation
  | SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder  for further details.
  | [2019-03-22 15:51:42,807][INFO ][container.run            ] Elasticsearch is ready and listening
  | /usr/share/elasticsearch/init ~
  | [2019-03-22 15:51:42,935][INFO ][container.run            ] Forcing the seeding of ACL documents
  | [2019-03-22 15:51:42,943][INFO ][container.run            ] Seeding the searchguard ACL index.  Will wait up to 604800 seconds.
  | [2019-03-22 15:51:43,049][INFO ][container.run            ] Seeding the searchguard ACL index.  Will wait up to 604800 seconds.
  | /etc/elasticsearch /usr/share/elasticsearch/init
  | Search Guard Admin v5
  | Will connect to localhost:9300 ... done
  | ERROR StatusLogger No Log4j 2 configuration file found. Using default configuration (logging only errors to the console), or user programmatically provided configurations. Set system property 'log4j2.debug' to show Log4j 2 internal initialization logging. See https://logging.apache.org/log4j/2.x/manual/configuration.html  for instructions on how to configure Log4j 2
  | Elasticsearch Version: 5.6.13
  | Search Guard Version: <unknown>
  | Contacting elasticsearch cluster 'elasticsearch' ...
  | Clustername: logging-es
  | Clusterstate: GREEN
  | Number of nodes: 1
  | Number of data nodes: 1
  | .searchguard index does not exists, attempt to create it ... done (0-all replicas)
  | Populate config from /opt/app-root/src/sgconfig/
  | Will update 'config' with /opt/app-root/src/sgconfig/sg_config.yml
  | SUCC: Configuration for 'config' created or updated
  | Will update 'roles' with /opt/app-root/src/sgconfig/sg_roles.yml
  | SUCC: Configuration for 'roles' created or updated
  | Will update 'rolesmapping' with /opt/app-root/src/sgconfig/sg_roles_mapping.yml
  | SUCC: Configuration for 'rolesmapping' created or updated
  | Will update 'internalusers' with /opt/app-root/src/sgconfig/sg_internal_users.yml
  | SUCC: Configuration for 'internalusers' created or updated
  | Will update 'actiongroups' with /opt/app-root/src/sgconfig/sg_action_groups.yml
  | SUCC: Configuration for 'actiongroups' created or updated
  | Done with success
  | /usr/share/elasticsearch/init
  | [2019-03-22 15:51:54,920][INFO ][container.run            ] Seeded the searchguard ACL index
  | [2019-03-22 15:51:54,922][INFO ][container.run            ] Disabling auto replication
  | /etc/elasticsearch /usr/share/elasticsearch/init
  | Search Guard Admin v5
  | Will connect to localhost:9300 ... done
  | ERROR StatusLogger No Log4j 2 configuration file found. Using default configuration (logging only errors to the console), or user programmatically provided configurations. Set system property 'log4j2.debug' to show Log4j 2 internal initialization logging. See https://logging.apache.org/log4j/2.x/manual/configuration.html  for instructions on how to configure Log4j 2
  | Elasticsearch Version: 5.6.13
  | Search Guard Version: <unknown>
  | Reload config on all nodes
  | Auto-expand replicas disabled
  | /usr/share/elasticsearch/init
  | [2019-03-22 15:51:58,824][INFO ][container.run            ] Updating replica count to 0
  | /etc/elasticsearch /usr/share/elasticsearch/init
  | Search Guard Admin v5
  | Will connect to localhost:9300 ... done
  | ERROR StatusLogger No Log4j 2 configuration file found. Using default configuration (logging only errors to the console), or user programmatically provided configurations. Set system property 'log4j2.debug' to show Log4j 2 internal initialization logging. See https://logging.apache.org/log4j/2.x/manual/configuration.html  for instructions on how to configure Log4j 2
  | Elasticsearch Version: 5.6.13
  | Search Guard Version: <unknown>
  | Reload config on all nodes
  | Update number of replicas to 0 with result: true
  | /usr/share/elasticsearch/init
  | [2019-03-22 15:52:02,653][INFO ][container.run            ] Adding index templates
  | [2019-03-22 15:52:02,896][INFO ][container.run            ] Create index template 'com.redhat.viaq-openshift-operations.template.json'
  | {"acknowledged":true}[2019-03-22 15:52:03,901][INFO ][container.run            ] Create index template 'com.redhat.viaq-openshift-orphaned.template.json'
  | {"acknowledged":true}[2019-03-22 15:52:04,492][INFO ][container.run            ] Create index template 'com.redhat.viaq-openshift-project.template.json'
  | {"acknowledged":true}[2019-03-22 15:52:05,031][INFO ][container.run            ] Create index template 'common.settings.kibana.template.json'
  | {"acknowledged":true}[2019-03-22 15:52:05,509][INFO ][container.run            ] Create index template 'common.settings.operations.orphaned.json'
  | {"acknowledged":true}[2019-03-22 15:52:05,994][INFO ][container.run            ] Create index template 'common.settings.operations.template.json'
  | {"acknowledged":true}[2019-03-22 15:52:06,492][INFO ][container.run            ] Create index template 'common.settings.project.template.json'
  | {"acknowledged":true}[2019-03-22 15:52:06,979][INFO ][container.run            ] Create index template 'org.ovirt.viaq-collectd.template.json'
  | {"acknowledged":true}[2019-03-22 15:52:07,327][INFO ][container.run            ] Finished adding index templates
  | [2019-03-22 15:52:07,335][INFO ][container.run            ] Starting init script: 0500-remove-index-patterns-without-uid
  | [2019-03-22 15:52:07,593][INFO ][container.run            ] Found 0 index-patterns to evaluate for removal
  | [2019-03-22 15:52:07,596][INFO ][container.run            ] Completed init script: 0500-remove-index-patterns-without-uid with 0 successful and 0 failed bulk requests
  | [2019-03-22 15:52:07,603][INFO ][container.run            ] Starting init script: 0510-bz1656086-remove-index-patterns-with-bad-title
  | [2019-03-22 15:52:07,844][INFO ][container.run            ] Found 0 index-patterns to remove
  | [2019-03-22 15:52:08,115][INFO ][container.run            ] Completed init script: 0510-bz1656086-remove-index-patterns-with-bad-title
  | [2019-03-22 15:52:08,124][INFO ][container.run            ] Starting init script: 0520-bz1658632-remove-old-sg-indices
  | [2019-03-22 15:52:08,604][WARN ][container.run            ] Found .searchguard setting 'index.routing.allocation.include._name' to be null
  | [2019-03-22 15:52:08,607][INFO ][container.run            ] Updating .searchguard setting 'index.routing.allocation.include._name' to be null
  | [2019-03-22 15:52:08,871][INFO ][container.run            ] Completed init script: 0520-bz1658632-remove-old-sg-indices
  | ~
```

but when I open kibana the state is red
any idea how I can debug the situation ?

**Comment by richm** [comment]
@martinbe1io 
- what version of ansible are you using?
- this issue is closed - please open another issue, probably at https://github.com/openshift/origin-aggregated-logging/issues since this doesn't appear to be an openshift-ansible problem

**Comment by odbawler** [comment]
@martinbe1io 
Judging by the error you got 

> "error: #1 is already in progress (Running).

The issue doesn't appear fixed, and you're probably using an ansible 2.7.x version. My guess would be the Ansible installer fails before it has finished configuring the Kibana installation. 

Ansible 2.6.x is still the only fix i'm aware of.

**Comment by martinbe1io** [comment]
obviously I'm using the wrong version but I thought that this issue is fixed now

```
[root@oc2n01de openshift-ansible]# ansible --version
ansible 2.7.5
  config file = /root/openshift_setup/openshift-ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Oct 30 2018, 23:45:53) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
```

and I would be right in #11132 but the topic seems to be stale... ;-(
